### ISSUE TITLE
Microsoft.Office.Interop.Excel not working with VS 2022 after 32 to 64-bit upgrade of Office

### ISSUE THREAD

**Issue by HKafeman** [MicrosoftDocs/OfficeDocs-DeployOffice#1264]
I used the Office Deployment Tool with the following in configuration.xml:
  OfficeClientEdition="64" MigrateArch="TRUE"

It upgraded Office okay, i.e. from:
  Microsoft® Excel® for Microsoft 365 MSO (Version 2307 Build 16.0.16626.20170) 32-bit
to:
  Microsoft® Excel® for Microsoft 365 MSO (Version 2308 Build 16.0.16731.20052) 64-bit

Excel, Word and Outlook are working fine.

BUT when trying to Build a VB.NET project with Visual Studio 2022 I now get these Errors:

> warning MSB3287: Cannot load type library for reference "Microsoft.Office.Interop.Excel". Error loading type library/DLL. (Exception from HRESULT: 0x80029C4A (TYPE_E_CANTLOADLIBRARY))

> warning BC40056: Namespace or type specified in the Imports 'Microsoft.Office.Interop.Excel' doesn't contain any public member or cannot be found. Make sure the namespace or the type is defined and contains at least one public member. Make sure the imported element name doesn't use any aliases._

The target dll is in place:
  C:\Windows\assembly\GAC_MSIL\Microsoft.Office.Interop.Excel\15.0.0.0__71e9bce111e9429c\Microsoft.Office.Interop.Excel.dll

I actually re-installed it using "Visual Studio 2010 Tools for Office Runtime" v10.0.60910 08/08/23 "vstor_redist.exe", but that made no difference, other than updating the file's date/time.

I have tried removing and re-installing Microsoft.Office.Interop.Excel in VS 2022, but that made no difference either!

In fact the VB.NET App built by my colleagues installs and runs fine on their PC and on a Laptop that has never had 32-bit Office.
BUT on my Laptop it show this Error Dialogue box:

> Unable to cast COM object of type  'Microsoft.Office.Interop.Excel.ApplicationClass' to interface type 'Microsoft.Office.Interop.Excel._Application'. This operation failed because the QueryInterface call on the COM component for the interface with IID '{00208D5-0000-0000-C000-000000000046}' failed due to the following error: Error loading type library/dLL. (Exception from HRESULT: 0x80029C4A (TYPE_E_CANTLOADLIBRARY))."

0x80029C4A (TYPE_E_CANTLOADLIBRARY) is the same Error that VS reports when trying to Build!

**I cannot find a way of fixing this, so has something been missed or done wrongly by the upgrade?**
  E.g. Does VS and the runtime environment think it is 32-bit instead of 64-bit DLL?

**How can I find out what is wrong and how can I resolve this?**

Any help/pointers/suggestions would be much appreciated as I am completely stuck!

Thanks
Henry Kafeman

**Comment by DHB-MSFT** [comment]
@HKafeman  For troubleshooting questions such as these, we recommend using one of these resources:

- [Microsoft Support](https://support.microsoft.com/contactus)
- [Microsoft Answers](https://answers.microsoft.com/)
- [Visual Studio support resources](https://visualstudio.microsoft.com/vs/support/#talktous)

**Comment by HKafeman** [comment]
Dan

Thank you for you quick reply.

I do not have any support subscription and only do minimal work work with VS!

Actually I just found this:

By removing this Registry Key, I got the built App to work okay!:
HKEY_CLASSES_ROOT\TypeLib\{00020813-0000-0000-C000-000000000046}\1.9\0\Win32

Which had:
(Default) = C:\Program Files (x86)\Microsoft Office\Root\Office16\EXCEL.EXE

  I.e. the original 32-bit path for Excel!

VS is still not building the App, but the with the Library are not there, instead, many instances of these Errors, etc.:

> BC30561	'xxxx' is ambiguous, imported from the namespaces or types 'System.Windows.Forms, Microsoft.Office.Interop.Excel'.
BC30002	Type 'xxxx' is not defined.
BC30590	Event 'xxxx' cannot be found.

So there is some problem with the References selected! - I wonder if there is some difference in which elements are included in 64-bit versus 32-bit libraries???

I will try to investigate further next week and find out if there are any differences in the References used by my colleagues.

Actually there are lots of references in the Registry to "C:\Program Files (x86)\Microsoft Office" which no longer exists!

Thanks
Henry Kafeman
